src/System.Collections.Concurrent/tests/ConcurrentDicionary/ConcurrentDictionaryExtensions.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Concurrent.Tests
+{
+    // Allows the ConcurrentDictionary tests to run on targets that do not have the new GetOrAdd/AddOrUpdate overloads.
```

Comment:
It wasn't as clean to separate out the tests into a separate file in this case. Steve suggested the extension methods. https://github.com/dotnet/corefx/pull/13828#discussion_r88785528